src/appkit.cs

```diff
@@ -27427,4 +27575,265 @@ interface NSStepperTouchBarItem
+	}
+
+	[Mac (10,15)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface NSTextCheckingController
+	{
+		[Export ("initWithClient:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (INSTextCheckingClient client);
+
+		[Export ("client")]
+		INSTextCheckingClient Client { get; }
+
+		[Export ("invalidate")]
+		void Invalidate ();
+
+		[Export ("didChangeTextInRange:")]
+		void DidChangeText (NSRange range);
+
+		[Export ("insertedTextInRange:")]
+		void InsertedText (NSRange range);
+
+		[Export ("didChangeSelectedRange")]
+		void DidChangeSelectedRange ();
+
+		[Export ("considerTextCheckingForRange:")]
+		void ConsiderTextChecking (NSRange range);
+
+		[Export ("checkTextInRange:types:options:")]
+		void CheckText (NSRange range, NSTextCheckingTypes checkingTypes, NSDictionary<NSString, NSObject> options);
+
+		[Export ("checkTextInSelection:")]
+		void CheckTextInSelection ([NullAllowed] NSObject sender);
+
+		[Export ("checkTextInDocument:")]
+		void CheckTextInDocument ([NullAllowed] NSObject sender);
+
+		[Export ("orderFrontSubstitutionsPanel:")]
+		void OrderFrontSubstitutionsPanel ([NullAllowed] NSObject sender);
+
+		[Export ("checkSpelling:")]
+		void CheckSpelling ([NullAllowed] NSObject sender);
+
+		[Export ("showGuessPanel:")]
+		void ShowGuessPanel ([NullAllowed] NSObject sender);
+
+		[Export ("changeSpelling:")]
+		void ChangeSpelling ([NullAllowed] NSObject sender);
+
+		[Export ("ignoreSpelling:")]
+		void IgnoreSpelling ([NullAllowed] NSObject sender);
+
+		[Export ("updateCandidates")]
+		void UpdateCandidates ();
+
+		[Export ("validAnnotations")]
+		string[] ValidAnnotations { get; }
+
+		[Export ("menuAtIndex:clickedOnSelection:effectiveRange:")]
+		[return: NullAllowed]
+		NSMenu MenuAtIndex (nuint location, bool clickedOnSelection, NSRange effectiveRange);
+
+		[Export ("spellCheckerDocumentTag")]
+		nint SpellCheckerDocumentTag { get; set; }
+	}
+
+	delegate NSCollectionViewItem NSCollectionViewDiffableDataSourceItemProvider (NSCollectionView collectionView, NSIndexPath indexPath, NSObject itemIdentifierType);
+
+	delegate NSView NSCollectionViewDiffableDataSourceSupplementaryViewProvider (NSCollectionView collectionView, string str, NSIndexPath indexPath);
+
+	[Mac (10,15)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface NSCollectionViewDiffableDataSource<SectionIdentifierType, ItemIdentifierType> : NSCollectionViewDataSource
+		where SectionIdentifierType : NSObject
+		where ItemIdentifierType : NSObject {
+
+		[Export ("initWithCollectionView:itemProvider:")]
+		IntPtr Constructor (NSCollectionView collectionView, NSCollectionViewDiffableDataSourceItemProvider itemProvider);
+
+		[Export ("snapshot")]
+		NSDiffableDataSourceSnapshot<SectionIdentifierType, ItemIdentifierType> Snapshot { get; }
+
+		[Export ("applySnapshot:animatingDifferences:")]
+		void ApplySnapshot (NSDiffableDataSourceSnapshot<SectionIdentifierType, ItemIdentifierType> snapshot, bool animatingDifferences);
```

Comment:
Apply sounds nicer, since we know is a Snapshot as the first parameter.

Reply:
Swift version is `applySnapshot` -> https://developer.apple.com/documentation/uikit/uitableviewdiffabledatasourcereference/3356326-applysnapshot

Arguably